### ISSUE TITLE
Vulkan API call failed with unrecoverable error: Out of pool memory

### ISSUE THREAD

**Issue by RayZina** [RPCS3/rpcs3#5424]
CPU: i7-6500U @ 2.50GHz
GPU: Intel Graphics HD 520
OS: Windows 10 64-bit
RPCS3 version: rpcs3-v0.0.5-7576-76f94940_win64

**Comment by AniLeo** [comment]
Intel drivers are not supported, but will wait for @kd-11 to triage this one

**Comment by 13xforever** [comment]
No log, no mention of system RAM amount. For all we know, you have 4 GB or RAM.
Also what happens when you disable Vulkan memory allocator in debug settings?

**Comment by mrcmunir** [comment]
I can confirm crash under linux driver too 
```
 RSX: St13runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Out of pool memory (VK_ERROR_OUT_OF_POOL_MEMORY_KHR
Returned by rpcs3/Emu/RSX/VK/VKGSRender.cpp:1058)
```
happens with any game sooner or later it has become unstable now.

Change   Disable Vulkan Memory Allocator to true same error.

**Comment by RayZina** [comment]
Hi. My apologies for missing out the game, log file and system ram. 
Game: Tales of Xillia[BLES01815]
System Ram: 16Gb
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2682493/RPCS3.log)

Disabling Vulkan memory allocator in debug settings still causes the same issue.

**Comment by kd-11** [comment]
I dont have intel hw to test, but this looks to be quite obvious. We request 8k descriptors and actually use all of them, but intel seemingly cannot handle this. They return success when requested, but later when trying to allocate the driver shits itself. This is rather unfortunate as there are games that emit a huge number of geometry calls and lowering the limit negatively affects other gpus. I'll see about a workaround, but with problems like these I'm not very hopeful for intel support.

**Comment by Fumiru** [comment]
CPU: i5-3450 @3.10GHz
GPU: AMD RX560 4GB
RAM: 4GB
OS: Windows 10 64-bit
RPCS3 version: v0.0.5-7587-5a40c180
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2687571/RPCS3.log.gz)

**Comment by jakemd7** [comment]
CPU: i3-6100 @3.70GHz
GPU: internal VGA (i530) [drivers; dch_win64_25.20.100.6471/win64_15.65.5.4982 --> same results!]
RAM: 8GB
OS: Windows 10 64-bit
RPCS3 version: v0.0.5-7536 (2018/11/30) ~ recent versions
                       --> error !!! (Vulkan API call failed with unrecoverable error: Out of pool memory)

                       but, roll-back to rpcs3-v0.0.5-7513 (2018/11/28) --> OK !!!)

https://github.com/RPCS3/rpcs3-binaries-win/releases/download/build-96cabeadff0c5a2052137a94f81c88912bcd5d44/rpcs3-v0.0.5-7513-96cabead_win64.7z

Thank U.

**Comment by kd-11** [comment]
Is this crash reproducible with the basic samples shipped with the emulator? (e.g gs_gcm_cube.elf or gs_gcm_basic_triangle.elf)

**Comment by mrcmunir** [comment]
@kd-11 yes with gs_gcm_cube.elf sucess easy after some time i.e 1-2minutes avg out of pool memory 
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2716735/RPCS3.log)

**Comment by reaper185** [comment]
Intel i7-7500u
Intel HD Graphics 620
16Gb Ram
250Gb Ssd
RPCS3 v0.0.5-7642-632b0f48

BLES02057 crash with vulkan at press start screen
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2718529/RPCS3.log)

**Comment by kd-11** [comment]
Should be fixed by https://github.com/RPCS3/rpcs3/pull/5509. Retest and close if fixed.

**Comment by mrcmunir** [comment]
@kd-11 I Retest in cube demo and some games working fine no issues detected it so it's fixed https://github.com/RPCS3/rpcs3/commit/a13986ec5cb7de2fdcb7e86dd9f4a10c3b336f8a

**Comment by jakemd7** [comment]
> Should be fixed by #5509. Retest and close if fixed.

It works !!! Thank U. [i3-6100/int. VGA(i530)]

**Comment by reaper185** [comment]
> 
> 
> Intel i7-7500u
> Intel HD Graphics 620
> 16Gb Ram
> 250Gb Ssd
> RPCS3 v0.0.5-7642-632b0f48
> 
> BLES02057 crash with vulkan at press start screen
> [RPCS3.log](https://github.com/RPCS3/rpcs3/files/2718529/RPCS3.log)

fixed
